services/statestorage/service.go

```diff
@@ -31,6 +31,7 @@ type service struct {
 }
 
 func NewStateStorage(config Config, persistence adapter.StatePersistence, reporting log.BasicLogger) services.StateStorage {
+	persistence.WriteMerkleRoot(0, merkle.GetEmptyNodeHash()) // TODO genesis block merkle here
```

Comment:
if in `merkle_forest.go` we interpreted a `nil` merkle root as if it was the empty trie hash we could avoid this initialization. we would gain:
* no need for a public `GetEmptyNodeHash()` (or rename it). it can remain encapsulated in `merkle_forest.go`
* we would not need to hard code the genesis block merkle root here. we could let the init flow just call `CommitStateDiff` for block height 0 and it would work as if the merkle root was hard coded.

the down side of this idea is that wherever we currently have `f.nodes[rootHash.KeyForMap()]` in `merkle_forest.go` we would need to replace with a call to a function that would check for `rootHash == nil` and replace it with the empty hash accordingly. this is less elegant on the merkle lib side. 

what do you think?

Reply:
found a common ground solution that is acceptable by majority of consensus